public/app/overrides/components/ExportData.tsx

```diff
@@ -1,15 +1,434 @@
+      flamebearer.metadata.appName,
+      flamebearer.metadata.startTime,
+      flamebearer.metadata.endTime
+    );
+    // get user input from modal
+    const customExportName = await getCustomExportName(defaultExportName);
+    // return if user cancels the modal
+    if (!customExportName) {
+      return;
+    }
+    // todo CORS
+    const response = await fetch('https://flamegraph.com/upload/v1', {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+      },
+      body: JSON.stringify({
+        fileTypeData: {
+          units: flamebearer.metadata.units,
+          spyName: flamebearer.metadata.spyName,
+        },
+        name: customExportName,
+        profile: btoa(JSON.stringify(flamebearer)),
+        type: 'json',
+      }),
+    });
+    const data = await response.json();
+    console.log(data);
+    // props.exportFlamegraphDotComFn(customExportName).then((url) => {
+    //   // there has been an error which should've been handled
+    //   // so we just ignore it
+    //   if (!url) {
+    //     return;
+    //   }
+
+    //   const dlLink = document.createElement('a');
+    //   dlLink.target = '_blank';
+    //   dlLink.href = url;
+
+    //   document.body.appendChild(dlLink);
+    //   dlLink.click();
+    //   document.body.removeChild(dlLink);
+    // });
+  };
+
+  const downloadPNG = async () => {
+    if (exportPNG) {
+      const { flamebearer } = props;
+
+      const defaultExportName = getFilename(
+        flamebearer.metadata.appName,
+        flamebearer.metadata.startTime,
+        flamebearer.metadata.endTime
+      );
+      // get user input from modal
+      const customExportName = await getCustomExportName(defaultExportName);
+      // return if user cancels the modal
+      if (!customExportName) {
+        return;
+      }
+
+      const filename = `${customExportName}.png`;
+
+      const mimeType = 'png';
+      // TODO use ref
+      // this won't work for comparison side by side
+      const canvasElement = document.querySelector(
+        '.flamegraph-canvas'
+      ) as HTMLCanvasElement;
+      const MIME_TYPE = `image/${mimeType}`;
+      const imgURL = canvasElement.toDataURL();
+      const dlLink = document.createElement('a');
+
+      dlLink.download = filename;
+      dlLink.href = imgURL;
+      dlLink.dataset.downloadurl = [
+        MIME_TYPE,
+        dlLink.download,
+        dlLink.href,
+      ].join(':');
+
+      document.body.appendChild(dlLink);
+      dlLink.click();
+      document.body.removeChild(dlLink);
+      setToggleMenu(!toggleMenu);
+    }
+  };
+
+  const handleToggleMenu = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setToggleMenu(!toggleMenu);
+  };
+
+  const downloadPprof = async function () {
+    if (!exportPprof) {
+      return;
+    }
+
+    if (props.exportPprof) {
+      // get user input from modal
+      const customExportName = await getCustomExportName('profile.pb.gz');
```

Comment:
I wonder if we could default to something better like from date

```
${YYYY-MM-DD HH:MM:SS}-${name}-${matchers}
```

My download folder already has profile.pb.gz 🙂

Reply:
Yes we can !